main.py

```diff
@@ -292,6 +293,30 @@ def disable_switch(self, request: Request) -> JSONResponse:
         self.notify_topology_update()
         self.notify_switch_links_status(switch, "link disabled")
         return JSONResponse("Operation successful", status_code=201)
+    
+    @rest('v3/switches/{dpid}', methods=['DELETE'])
+    def delete_switch(self, request: Request) -> JSONResponse:
+        """Delete switch"""
+        dpid = request.path_params["dpid"]
+        with self._switch_lock:
+            try:
+                switch: Switch = self.controller.switches[dpid]
+            except KeyError:
+                raise HTTPException(404, detail="Switch not found.")
+            if switch.status != EntityStatus.DISABLED:
+                raise HTTPException(409, detail=f"The switch is not disable.")
+            links = self.links.copy()
+            for link_id, link in links.items():
+                if link.endpoint_a.switch.dpid == dpid or link.endpoint_b.switch.dpid == dpid:
+                    raise HTTPException(409, detail=f"This switch have link {link_id}")
+            switch = self.controller.switches.pop(dpid)
+            self.topo_controller.delete_switch(dpid)
```

Comment:
Another point, what about `interface_details` available tags and tag ranges? I believe you need to ensure that all interfaces of a switch still have the default values in order to be able to delete it, otherwise it can leak persistent data, especially after restarts. Similarly to requiring deleted flows, here we need to make sure that NApps are also deallocating correctly all vlans.

Reply:
Now it is checking every `available_tags` against `tag_ranges` from each Interface. Does it every interface `tag_lock`?

Reply:
Good point. Yes, to be thread safe here you'd need `interface._tag_lock`, same way as you've structured on Kytos core. `_tag_lock` being a "private" lock shouldn't be accessed directly here, but, you can introduce it in the meantime, and then later on please a PR on kytos core introducing a new method on the interface to make this check. Also, on `switch.interfaces.items()` you want a shallow copy since a new interface can get discovered via openflow in some cases can change that dict size mid iteration (extremely rare to happen but it can happen).

Reply:
Now using `Interface.all_tags_available()` introduced in kytos [PR 436](https://github.com/kytos-ng/kytos/pull/436)